pyopensn/src/aquad.cpp

```diff
@@ -0,0 +1,129 @@
+// SPDX-FileCopyrightText: 2025 The OpenSn Authors <https://open-sn.github.io/opensn/>
+// SPDX-License-Identifier: MIT
+
+#include "pyapi.hpp"
+
+#include <memory>
+#include <stdexcept>
+
+#include "framework/math/quadratures/angular/angular_quadrature.h"
+#include "framework/math/quadratures/angular/product_quadrature.h"
+#include "framework/math/quadratures/angular/sldfe_sq_quadrature.h"
+#include "framework/logging/log.h"
+
+namespace opensn {
```

Comment:
For the failing clang-format check. We use the `.clang-format` file from the root of the repo, so you want to set your IDE to use that one. You are most likely using something else. 

Also, make sure that you are using clang-format-14 from your IDE (IIRC there is a plugin for this where you can specify the path to the clang-format binary). It is ok to have different clang-format version and different compiler version (if you are using newer llvm). The problem is that newer versions of clang-format can produce different formatting with the same `.clang-format` file. The version 14 is what the CI uses, so this is why we recommend to use that locally, so you don't have to fight with it.